### PR TITLE
Add appToken refreshing to requests; require token definition

### PR DESCRIPTION
**Pull request by mlingner**
Addresses #64 and requires token definitions for all calls. This would be a major revision. Please do not merge yet.

### CODE REVIEW

**Comment by mlingner** [comment]
Hey guys, please review my PR for a major revision to classy-node. This is not fully tested (yet) and will require some work on Appv2. Those are things I plan to do tonight and tomorrow, but for now it would be great to get your feedback on what I've done here.

The major changes are as follows:

1. App tokens will now be checked and a refresh will be attempted for every request that wants to use an app token.
2. Member tokens are no longer managed by classy-node. It is instead required that they be passed in.
3. Related to 2, all requests using classy-node must include a token definition or `token: app`

Thanks!

File: src/ClassyResource.js [hunk 1]
```diff
@@ -75,46 +73,106 @@ export default class ClassyResource {
...
+  /**
+   * If this._classy.appToken exists, just resolve,
+   * if not, refresh and then resolve.
+   * @return {[type]} [description]
+   */
+  _refreshAppToken() {
+    const promise = new Promise((resolve, reject) => {
+      if (!this._classy.appToken) {
+        this._classy.getAppToken().then((response) => {
+          this._classy.setAppToken(respnose);
```
Comment (tomborger) [inline src/ClassyResource.js]: respnose

**Comment by mlingner** [comment]
Merging and updating version so TK and I can get started on Appv2 work. If you have time, please do review in case there are errors I missed.

File: src/ClassyResource.js [hunk 2]
```diff
@@ -75,46 +73,106 @@ export default class ClassyResource {
...
+        this._classy.getAppToken().then((response) => {
+          this._classy.setAppToken(response);
+          resolve(response);
+        }, (error) => {
+          reject(error);
+        });
+      } else {
+        resolve(this._classy.appToken);
+      }
+    });
```
Comment (tomborger) [inline src/ClassyResource.js]: It seems to me you could still have concurrency issues here if an app token expires and then multiple requests come in simultaneously, each fetch a new app token, then overwrite each other with the result, causing all but one to fail. Might want to have a global "pendingAppToken" flag/promise that you can check and/or .then() off of if appToken is null.

**Pull request merged**